### PR TITLE
Layouts/bootstrap

### PR DESCRIPTION
**Pull request by tehreemtalat7**

### CODE REVIEW

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -1,27 +1,108 @@
...
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="nav navbar-nav ml-auto">
+        <li class="nav-item active">
+          <%= link_to "Home", menus_index_path, class: 'nav-link' %>
+        </li>
+      </ul>
+
+      <ul class="navbar-nav ml-auto">
+
+        <!-- Attendance Button -->
```
Comment (mujeeb-arshad) [inline app/views/layouts/_header.html.erb]: remove

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -1,27 +1,108 @@
...
+          <div class="d-flex align-items-center">
+            <% if present_marked? && !logged_out? %>
+              <%= link_to 'Update Logout', update_attendance_path, class: 'btn btn-sm btn-outline-danger update_attendance', remote: true %>
+            <% elsif !present_marked? %>
+              <%= link_to 'Mark Present', employee_attendance_path, class: 'btn btn-sm btn-regular btn-primary mark_present', remote: true %>
+             <% end %>
+          </div>
+        </li>
+
+        <!-- Notifications -->
```
Comment (mujeeb-arshad) [inline app/views/layouts/_header.html.erb]: remove

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -1,27 +1,108 @@
...
+        <li class="nav-item dropdown no-arrow">
+          <a class="nav-link dropdown-toggle" href="#" id="alertsDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+            <i class="fas fa-bell fa-fw"></i>
+            <sup>9+</sup>  
+          </a>
+
+          <!-- Notification Dropdown-->
```
Comment (mujeeb-arshad) [inline app/views/layouts/_header.html.erb]: remove

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -1,27 +1,108 @@
...
+                  <i class="fas fa-file-alt text-white"></i>
+                </div>
+              </div>
+              <span class="small-gray">A new report is here!</span>
+            </a>
+          </div>
+
+        </li>
+
+        <!-- Messages -->
```
Comment (mujeeb-arshad) [inline app/views/layouts/_header.html.erb]: remove

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -1,27 +1,108 @@
...
+        <li class="nav-item dropdown no-arrow">
+          <a class="nav-link dropdown-toggle" href="#" id="messagesDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+            <i class="fas fa-envelope fa-fw"></i>
+            <sup>7</sup>
+          </a>
+
+          <!-- Messages Dropdown-->
```
Comment (mujeeb-arshad) [inline app/views/layouts/_header.html.erb]: remove

File: app/views/layouts/_header.html.erb [hunk 6]
```diff
@@ -1,27 +1,108 @@
...
+                </div>
+              </div>
+              <span class="small-gray">New Message!</span>
+            </a>
+          </div>
+        </li>
+
+        <div class="topbar-divider d-none d-sm-block"></div>
+
+        <!-- User -->
```
Comment (mujeeb-arshad) [inline app/views/layouts/_header.html.erb]: remove

File: app/views/layouts/_header.html.erb [hunk 7]
```diff
@@ -1,27 +1,108 @@
...
+        <li class="nav-item dropdown no-arrow">
+          <a class="nav-link dropdown-toggle" href="#" id="userDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+            <span class="small-gray"><%= current_employee.name %></span>
+            <img src="https://source.unsplash.com/QAB-WJcbgJk/60x60" class="img-profile rounded-circle">
+          </a>
+
+          <!-- User option Dropdown -->
```
Comment (mujeeb-arshad) [inline app/views/layouts/_header.html.erb]: remove

File: app/views/layouts/_sidebar.html.erb [hunk 8]
```diff
@@ -1,113 +1,64 @@
-<!-- Sidebar -->
```
Comment (mujeeb-arshad) [inline app/views/layouts/_sidebar.html.erb]: remove

File: app/views/layouts/_sidebar.html.erb [hunk 9]
```diff
@@ -1,113 +1,64 @@
...
-    </div>
-    <div class="sidebar-brand-text mx-3">SB Admin <sup>2</sup></div>
-  </a>
+<% @sidebar_data = generate_sidebar_options %>
 
-  <!-- Divider -->
-  <hr class="sidebar-divider my-0">
+<nav id="sidebar">
 
-  <!-- Nav Item - Dashboard -->
```
Comment (mujeeb-arshad) [inline app/views/layouts/_sidebar.html.erb]: remove comments throughout the page

File: app/views/projects/index.html.erb [hunk 10]
```diff
@@ -2,9 +2,9 @@
...
 </div>
 <h1>Listing Projects</h1>
-<%= link_to 'New Project', new_project_path, class: "btn btn-primary" %>
+<%= link_to 'New Project', new_project_path, class: "btn btn-regular btn-primary" %>
 <% if !@projects.nil?%>								
-<div class="row">
+<div class="row" style="margin-top: 40px;">
 	<% @projects.each do |project| %>
 	<tbody>
 		<div class="card col-sm-4">
```
Comment (mujeeb-arshad) [inline app/views/projects/index.html.erb]: properly indent

File: app/controllers/attendances_controller.rb [hunk 11]
```diff
@@ -29,13 +26,11 @@ def create
...
-    @employees_todays_attendance = current_employee.todays_attendance_of_employee(@company)
-    
+    @employees_todays_attendance = current_employee.todays_attendance_of_employee
+
     unless @employees_todays_attendance.blank?
-      @employee_attendance = @company.attendances.find_by(employee_id: @current_employee.id)
-      @employee_attendance.logout_time = DateTime.now
-      @employee_attendance.save!
+      @employees_todays_attendance.logout_time = DateTime.now
+      @result = @employees_todays_attendance.save!
```
[the remark this answers is not in the document]
Comment (tehreemtalat7) [reply app/controllers/attendances_controller.rb]: This is handled in the JS file (update.js.erb)

File: app/helpers/application_helper.rb [hunk 12]
```diff
@@ -1,2 +1,119 @@
+require 'attendance'
+
 module ApplicationHelper
+  def present_marked?
+    @attendance = current_employee.todays_attendance_of_employee
+    !@attendance.blank?
```
Comment (zainsaleem4338) [inline app/helpers/application_helper.rb]: we can simply use "@attendance.present?"

File: app/helpers/application_helper.rb [hunk 13]
```diff
@@ -1,2 +1,119 @@
...
+  end
+
+  def logged_out?
+    @attendance = current_employee.todays_attendance_of_employee
+    @attendance.logout_time
```
Comment (zainsaleem4338) [inline app/helpers/application_helper.rb]: is logout_time boolean?

File: app/helpers/application_helper.rb [hunk 14]
```diff
@@ -1,2 +1,119 @@
...
+  end
+
+  def admin?
+    current_employee.role.eql? 'Admin'
```
Comment (zainsaleem4338) [inline app/helpers/application_helper.rb]: use 'Admin' constant

File: Gemfile [hunk 15]
```diff
@@ -41,7 +39,6 @@ group :test do
   gem 'better_errors'
   gem 'pry', '=0.10.1'
   gem 'pry-rails', '=0.3.3'
-  gem 'rspec-rails', '~> 3.8'
```
Comment (zainsaleem4338) [inline Gemfile]: why are we removing this gem from the test group?

File: app/views/attendances/_attendance_table.html.erb [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+      <th>Employee id</th>
+      <th>Name</th>
+      <th>Login Time</th>
+      <th>Logout Time</th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @attendances_list.each do |attendance| %>
+    <% @current_employee = Employee.find(attendance.employee_id) %>
```
Comment (zainsaleem4338) [inline app/views/attendances/_attendance_table.html.erb]: why are we changing current_employee here?

File: Gemfile (hunk 15 above)
Comment (tehreemtalat7) [reply Gemfile]: its was present in both groups and outside

File: app/helpers/application_helper.rb (hunk 12 above)
Comment (tehreemtalat7) [reply app/helpers/application_helper.rb]: done, using a function 'present?' in Attendance model

File: app/helpers/application_helper.rb (hunk 13 above)
Comment (tehreemtalat7) [reply app/helpers/application_helper.rb]: no, its nil or Datetime

Comment (tehreemtalat7) [reply app/helpers/application_helper.rb]: made it logout_time.blank?

File: app/helpers/application_helper.rb (hunk 14 above)
Comment (tehreemtalat7) [reply app/helpers/application_helper.rb]: done

**Pull request merged**